web_dwc2.py

```diff
@@ -623,467 +613,251 @@ def rr_reply(self, web_):
+            for name_ in temp_names:
+                self.status["temps"]["names"].append(name_)
 
         if self.message:
-            self.status_3.update(
+            self.status.update(
                 self.message
             )
 
         if self.popup:
-            self.status_3.update(
+            self.status.update(
                 self.popup
             )
 
-        if self.chamber:
-            chamber_stats = self.chamber.get_status(0)
-            self.status_3['temps'].update({
-                "chamber": {
-                    "current": chamber_stats.get("temp"),
-                    "active": chamber_stats.get("target", -1),
-                    "state": chamber_stats.get("state"),
-                    "heater": 2,  # extruders + bett ?
+        if (rr_status_type == 2):
+            self.status.update({
+                "coldExtrudeTemp": ex_min_extrude_temp,
+                "coldRetractTemp": ex_min_extrude_temp,
+                "compensation": "None",
+                "controllableFans": len(fan_stats),
+                "tempLimit": ex_max_temp,
+                "endstops": 4088,                                                       # TODO: calc entstops
+                "firmwareName": "Klipper",
+                "geometry": self.kin_name,
+                "axes": len(self.get_axes_homed()),
+                "totalAxes": len(self.get_axes_homed()) + len(self.extruders),
+                "axisNames": "XYZ",
+                "volumes": 1,
+                "mountedVolumes": 1,
+                "name": self.printername,
+                "probe": {
+                    "threshold": 100,
+                    "height": 0,
+                    "type": 8
                 },
-                "current": self.status_3['temps']['current'] + [chamber_stats.get("temp")],
-                "state": self.status_3['temps']['state'] + [chamber_stats.get("state")],
-                "names": self.status_3['temps']['names'] + ["Chamber"],
-                "extra": self.status_3['temps']['extra'] + [
-                    {
-                        "name": "Chamber Fan Speed",
-                        "temp": self.chamber.get_speed()
-                    }
-                ]
+                "tools": [],
             })
 
-        if self.get_host_temp() > 0:
-            self.status_3['temps'].update({
-                "extra": self.status_3['temps']['extra'] + [
-                    {
-                        "name": "Host",
-                        "temp": self.get_host_temp()
-                    }
-                ]
+            if self.extruders is not None:
+                index_ = 0
+                for ex_ in self.extruders:
+                    if ex_ is not None:
+                        self.status["tools"].append({
+                            "number": index_,
+                            "name": "Tool" + str(index_),
+                            "heaters": [index_ + 1],  # TODO: Can Klipper has more heaters per tool
```

Comment:
This might be correct theese days. We had once a hardtime to fidle were to +1 and were not in #landofschnitzel. So this needs intense testing with toolselection etc.

Reply:
because headbed has ID 0 and the heaters/extruders start with ID 1. this will link the tool to the heater.

Reply:
wouldn´t make my sign here without testing it on multiextruder system. We had that see the ticket. #7